src/uu/expand/src/expand.rs

```diff
@@ -431,12 +428,18 @@ fn expand(options: &Options) -> std::io::Result<()> {
                         // now dump out either spaces if we're expanding, or a literal tab if we're not
                         if init || !options.iflag {
                             if nts <= options.tspaces.len() {
-                                output.write_all(options.tspaces[..nts].as_bytes())?;
+                                output
+                                    .write_all(options.tspaces[..nts].as_bytes())
+                                    .map_err_context(|| "failed to write output".to_string())?;
```

Comment:
I don't love all of these identical calls to `map_err_context` I added here and below, this was just an attempt to keep the error output messages consistent with what they currently. I can think of a few options here:

1) I'm not sure if "failed to write output" is necessary to keep the output consistent with the GNU version of `expand`, but if not then maybe we could just drop these.

2) I can think of a few ways we could potentially refactor this part of the function to reduce the identical `map_err_context` calls, including maybe moving all the logic that handles writing the output into its own function that returns `std::io::Result<()>`, then we can just call `map_err_context` on that result.

3) If maybe I'm just overthinking things and what I currently have is fine, then maybe we can just leave it?

I think (2) seems like a decent idea, but I was initially trying to avoid doing anything too drastic here since this will be my first contribution to the project. But if people think something along these lines is worth it, I'd be happy to do it!

Reply:
I like option 2! Maybe you could also refactor that `while match` away (for example put that match in a separate function as well). It's a bit hard to follow at the moment.

Reply:
Alrighty, I moved the code that handles expanding a single line into its own function (`expand_line`), which encapsulates all of the "failed to write output" errors.